### PR TITLE
waypoint: init at 0.1.5

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Init waypoint at `0.1.5`
Based on @winpat's pr #100994
I'm happy for any of my modifications to be coppied over & delete this branch. I'll put a PR in upstream so we can probably use `make bin` as is.

Resolves the injection issue issue

nixpkgs_waypoint

Removed the dontPatchELF stuff because none of it seems to actually be passed through by `buildGoModule` and I'm not comfortable changing it.

###### Things done

I ran the init and the build but it could do with more testing.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winpat** [comment]
Great job! I closed my PR.

Should be able to test this tomorrow.

**Comment by winpat** [comment]
LGTM. All the issues I had with my original expressions are resolved.

**Review by winpat — APPROVED** [review]

**Comment by 06kellyjac** [comment]
It's looking like `-static` isn't going to be added upstream.
I've tried a bunch of things to get it working without it but no luck so far.

I'm also happy to add back in the git stuff for setting the version if you'd like

**Comment by winpat** [comment]
> I'm also happy to add back in the git stuff for setting the version if you'd like

That sounds like a good idea.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103754` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waypoint</li>
  </ul>
</details>

**Comment by 06kellyjac** [comment]
@winpat actually it looks like `fetchgit` with `leaveDotGit` isn't a great idea in its current state #100498 https://github.com/NixOS/nixpkgs/pull/64319#issuecomment-729837272
TIL

**Comment by 06kellyjac** [comment]
Also it doesn't look like there's going to be much progress investigating the issue so I'm happy with it as is & I don't want to keep you and others waiting by leaving this in review if it's working well with everything we've tried so far

**Review by vikanezrimaya — APPROVED** [review]
First of all, let me tell that I love reviewing Nixpkgs PRs because sometimes I discover a gem of a package that I instantly fall in love with the moment I see it. **This** is one of these gems.

I've read and followed the tutorial with a Node app while using Waypoint built out of this derivation, and it was seamless. No errors, it works perfectly. LGTM.

/status needs_merger

File: pkgs/applications/networking/cluster/waypoint/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  subPackages = ["."];
+
+  nativeBuildInputs = [ go-bindata ];
+
+  buildPhase = ''
+    CGO_ENABLED=0 go build -ldflags '-w -extldflags "-static"' -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint
+    cd internal/assets
+    go-bindata -pkg assets -o prod.go -tags assetsembedded ./ceb
+    cd ../../
+    CGO_ENABLED=0 go build -ldflags '-X github.com/hashicorp/waypoint/version.GitDescribe=v${version}' -tags assetsembedded -o ./waypoint ./cmd/waypoint
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/waypoint/default.nix:26]:
```suggestion
    CGO_ENABLED=0 go build -ldflags '-s -w -X github.com/hashicorp/waypoint/version.GitDescribe=v${version}' -tags assetsembedded -o ./waypoint ./cmd/waypoint
```

File: pkgs/applications/networking/cluster/waypoint/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  deleteVendor = true;
+  vendorSha256 = "1xdari6841jp6lpjwydv19v3wafj17hmnwsa2b55iw6dysm4yxdr";
+
+  subPackages = ["."];
+
+  nativeBuildInputs = [ go-bindata ];
+
+  buildPhase = ''
+    CGO_ENABLED=0 go build -ldflags '-w -extldflags "-static"' -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/waypoint/default.nix:22]:
```suggestion
    CGO_ENABLED=0 go build -ldflags '-s -w -extldflags "-static"' -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint
```

File: pkgs/applications/networking/cluster/waypoint/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cd internal/assets
+    go-bindata -pkg assets -o prod.go -tags assetsembedded ./ceb
+    cd ../../
+    CGO_ENABLED=0 go build -ldflags '-X github.com/hashicorp/waypoint/version.GitDescribe=v${version}' -tags assetsembedded -o ./waypoint ./cmd/waypoint
+    CGO_ENABLED=0 go build -tags assetsembedded -o ./waypoint-entrypoint ./cmd/waypoint-entrypoint
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/waypoint/default.nix:27]:
```suggestion
    CGO_ENABLED=0 go build -ldflags '-s -w' -tags assetsembedded -o ./waypoint-entrypoint ./cmd/waypoint-entrypoint
```

**Comment by 06kellyjac** [comment]
Added the extra flags to reduce size as requested.
Results:

```
before:
103.4 MB - waypoint
 45.4 MB - waypoint-entrypoint

after:
 90.7 MB - waypoint
 40.3 MB - waypoint-entrypoint
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103754` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waypoint</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103754` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**